src/EditorFeatures/CSharpTest/Intents/GenerateConstructorIntentTests.cs

```diff
@@ -62,6 +62,60 @@ public C(int someInt)
             await VerifyExpectedTextAsync(WellKnownIntents.GenerateConstructor, initialText, expectedText).ConfigureAwait(false);
         }
 
+        [Fact]
+        public async Task GenerateConstructorTypedPrivateWithIntentData()
+        {
+            var initialText =
+@"class C
+{
+    private readonly int _someInt;
+
+    {|typed:private C|}
+}";
+            var expectedText =
+@"class C
+{
+    private readonly int _someInt;
+
+    private C(int someInt)
+    {
+        _someInt = someInt;
+    }
+}";
+
+            // lang=json
+            var intentData = @"{ ""accessibility"": ""private""}";
```

Comment:
by the way syntax highlighting for this is great, thanks @CyrusNajmabadi

Reply:
:)